src/pool/cli.c

```diff
@@ -1850,8 +1886,8 @@ pool_query_cb(tse_task_t *task, void *data)
 		D_INFO("retry with map buffer size required by server (%ul)\n",
 		       out_v5->pqo_map_buf_size);
 		pool->dp_map_sz = out_v5->pqo_map_buf_size;
-		rc = tse_task_reinit(task);
-		D_GOTO(out, rc);
+		reinit          = true;
+		D_GOTO(out, rc = 0);
```

Comment:
interesting approach to set rc = 0 here, and also in the -DER_AGAIN reinit=true case below. Is this just for clarity / no potential confusion reading the code? When we get to the out label, with reinit=true, rc will be reassigned to the result of the task reinit call.

Reply:
Yes, I had exactly the same thought when authoring this change. And yes, the unnecessary `rc = 0` is for to express the intention that "reinitializing the task implies that we want this function to return 0 instead of `-DER_TRUNC`".